src/components/ModelingSidebar/ModelingPanes/MemoryPane.tsx

```diff
@@ -81,31 +81,64 @@ export const MemoryPane = () => {
 }
 
 export const processMemory = (variables: VariableMap) => {
-  const processedMemory: any = {}
+  const processedMemory: Record<
+    string,
+    string | number | boolean | object | undefined
+  > = {}
   for (const [key, val] of Object.entries(variables)) {
     if (val === undefined) continue
-    if (
-      val.type === 'Sketch' ||
-      // @ts-ignore
-      val.type !== 'Function'
-    ) {
-      const sk = sketchFromKclValueOptional(val, key)
-      if (val.type === 'Solid') {
-        processedMemory[key] = val.value.value.map(
-          ({ ...rest }: ExtrudeSurface) => {
-            return rest
-          }
-        )
-      } else if (!(sk instanceof Reason)) {
-        processedMemory[key] = sk.paths.map(({ __geoMeta, ...rest }: Path) => {
+    const sk = sketchFromKclValueOptional(val, key)
+    if (val.type === 'Solid') {
+      processedMemory[key] = val.value.value.map(
+        ({ ...rest }: ExtrudeSurface) => {
           return rest
-        })
+        }
+      )
+    } else if (!(sk instanceof Reason)) {
+      processedMemory[key] = sk.paths.map(({ __geoMeta, ...rest }: Path) => {
+        return rest
+      })
+    } else if (val.type === 'Function') {
+      processedMemory[key] = '__function__'
+    } else if (val.type === 'Number') {
+      // The type is wrong because ts-rs can't handle tuple types.
+      let ty: string = val.ty.type
+      // Map from UnitLen to NumericSuffix.
```

Comment:
I think `format_number` (or a better API) should handle types directly rather than requiring this mapping back to suffixes, since we lose information here.

Reply:
I split up the API in to `format_number_literal` and `human_display_number`. Check out `fmt.rs`.